proposals/0330-collection-conditionals.md

```diff
@@ -58,4 +217,10 @@ N/A. This is not an API.
 
 ## Alternatives considered
 
-It was decided to tackle this limited scope for the introduction of conditional syntax first, as specific use cases can be thought of and to be honest this has always seemed like a bit of an omission. Other areas where conditionals could be introduced abound but can be discussed with reference to their own particular subtleties of implementation separately at a later date.
+### Lexer based preprocessing
+
+Currently, processing of conditional compilation directives in Swift is based on parse trees. This proposal continues that approach.
+
+There has been discussion of achieving the goals of this proposal as well as unlocking many other possible uses of `#if` by switching Swift to a Lexer-based preprocessing model.
+
+A lexer-based solution would have significant impact on the current syntax tooling (e.g. `swift-format`). Most notably, a lexer-based model would require source tooling to have knowledge of the build arguments to be used to perform syntax analysis. Therefore, this proposal does not consider it, and focuses on expanding the current model to collection literals. It also does not close off the possibility of a lexer-based approach from being proposed at a later time, but that should be done as a separate initiative to what is proposed here.
```

Comment:
Looking over the review feedback, several people presented a related argument along the lines of “If we switch to a lexer-based model later, this proposal will create technical debt that will need to be cleaned up”. I think it may be a good idea to add a sentence or two discussing this—perhaps simply pointing out that the updated implementation is pretty small so it seems like there won’t be much tech debt to clean up.